src/test/java/au/org/ala/names/index/TaxonomyTest.java

```diff
@@ -0,0 +1,327 @@
+        this.taxonomy = new Taxonomy(config, null);
+        this.taxonomy.begin();
+        CSVNameSource source1 = new CSVNameSource(this.resourceReader("taxonomy-8.csv"));
+        this.taxonomy.load(Arrays.asList(source1));
+        this.taxonomy.resolve();
+        TaxonConceptInstance i11 = this.taxonomy.getInstance("NZOR-4-118018");
+        TaxonConceptInstance i12 = this.taxonomy.getInstance("urn:lsid:catalogueoflife.org:taxon:e5c9a6a6-e319-11e5-86e7-bc764e092680:col20161028");
+        TaxonConceptInstance i13 = this.taxonomy.getInstance("ALA_Closteroviridae");
+        assertNotNull(i11);
+        assertNotNull(i12);
+        assertNotNull(i13);
+        TaxonConcept tc1 = i11.getContainer();
+        assertSame(i11, tc1.getRepresentative());
+        assertSame(tc1, i12.getContainer());
+        assertSame(tc1, i13.getContainer());
+        assertSame(i11, i11.getResolved());
+        assertSame(i11, i11.getResolvedAccepted());
+        assertSame(i11, i12.getResolved());
+        assertSame(i11, i12.getResolvedAccepted());
+        assertSame(i11, i13.getResolved());
+        assertSame(i11, i13.getResolvedAccepted());
+        assertNull(i11.getResolvedParent());
+        assertNull(i12.getResolvedParent());
+        assertNull(i13.getResolvedParent());
+    }
+
+    @Test
+    public void testWrite1() throws Exception {
+        TaxonomyConfiguration config = TaxonomyConfiguration.read(this.resourceReader("taxonomy-config-2.json"));
+        this.taxonomy = new Taxonomy(config, null);
+        this.taxonomy.begin();
+        CSVNameSource source1 = new CSVNameSource(this.resourceReader("taxonomy-3.csv"));
+        CSVNameSource source2 = new CSVNameSource(this.resourceReader("taxonomy-4.csv"));
+        this.taxonomy.load(Arrays.asList(source1, source2));
+        this.taxonomy.resolve();
+        File dir = new File(this.taxonomy.getWork(), "output");
+        dir.mkdir();
+        this.taxonomy.createDwCA(dir);
+        assertTrue(new File(dir, "meta.xml").exists());
+        assertTrue(new File(dir, "eml.xml").exists());
+        assertTrue(new File(dir, "taxon.txt").exists());
+        assertTrue(new File(dir, "taxonvariant.txt").exists());
+        assertTrue(new File(dir, "identifier.txt").exists());
+        assertTrue(new File(dir, "rightsholder.txt").exists());
+
+    }
+
+
+    @Test
+    public void testWrite2() throws Exception {
+        TaxonomyConfiguration config = TaxonomyConfiguration.read(this.resourceReader("taxonomy-config-2.json"));
+        this.taxonomy = new Taxonomy(config, null);
+        this.taxonomy.begin();
+        CSVNameSource source1 = new CSVNameSource(this.resourceReader("taxonomy-10.csv"));
+        this.taxonomy.load(Arrays.asList(source1));
+        this.taxonomy.resolve();
+        File dir = new File(this.taxonomy.getWork(), "output");
+        dir.mkdir();
+        this.taxonomy.createDwCA(dir);
```

Comment:
DwcaNameIndexer has 0% test coverage right now. As a quick start, could you take this method and create a new test just to verify that the index is generated and can be read by Lucene, without analysing the contents of the Lucene index at this stage.